lib/src/kbasesearchengine/events/handler/WorkspaceEventHandler.java

```diff
@@ -570,4 +573,108 @@ private static String toWSRefPath(final List<GUID> objectRefPath) {
+        command = new HashMap<>();
+        command.put("command", "listObjects");
+        command.put("params", new ListObjectsParams().withShowOnlyDeleted(wsid).withIds(Arrays.asList(wsid)));
+        final List<Tuple11> deletedObjList;
+        try {
+            deletedObjList = ws.getClient().administer(new UObject(command))
+                    .asClassInstance(List.class);
+
+            for( Tuple11 obj: deletedObjList) {
+                long deletedObjid = Long.valueOf(obj.getE1().toString()).longValue();
+                if(deletedObjid  == objid) {
+                    latestEventType = StatusEventType.DELETED;
+                    break;
+                }
+            }
+        } catch (IOException e) {
+            throw new RetriableIndexingException(e.getMessage());
+        } catch (JsonClientException e) {
+            throw new RetriableIndexingException(e.getMessage());
+        }
+
+
+        // check if name has changed and update state of lastestName
+        ObjectSpecification os = new ObjectSpecification().withRef(
+                ev.getAccessGroupId().get().toString() + "/" +
+                        ev.getAccessGroupObjectId().get());
+
+        final List<ObjectSpecification> getInfoInput = new ArrayList<>();
+        getInfoInput.add(os);
+
+        command = new HashMap<>();
+        command.put("command", "getObjectInfo");
+        command.put("params", new GetObjectInfo3Params().withObjects(getInfoInput));
+
+        final GetObjectInfo3Results objInfo;
+        try {
+            objInfo = ws.getClient().administer(new UObject(command))
+                    .asClassInstance(GetObjectInfo3Results.class);
+            String name = objInfo.getInfos().get(0).getE2();
+            if ( !Utils.isNullOrEmpty(name) ) {
+                latestName = name;
+            }
+        } catch (IOException e) {
+            throw handleException(e);
+        } catch (JsonClientException e) {
+            throw handleException(e);
+        }
+
+        // check if permissions have changed and update state of isPublic
+        try {
+            command = new HashMap<>();
+            command.put("command", "getWorkspaceInfo");
+            command.put("params", new WorkspaceIdentity()
+                    .withId((long) ev.getAccessGroupId().get()));
+
+            final String isPublic;
+
+            isPublic = ws.getClient().administer(new UObject(command))
+                    .asClassInstance(WS_INFO_TYPEREF).getE6();
+
+            if ( !Utils.isNullOrEmpty(isPublic) ) {
+                latestIsPublic = (isPublic == "n") ? true: false;
+
+            }
+        } catch (IOException | JsonClientException ex) {
+            throw new RetriableIndexingException(ex.getMessage());
+        }
+
+        StatusEvent updatedEvent = StatusEvent.
+                getBuilder(ev.getStorageObjectType().get(), ev.getTimestamp(), latestEventType).
+                withNullableAccessGroupID(ev.getAccessGroupId().get()).
```

Comment:
May not have accgrp id, object id, or version - use .orNull()

Reply:
Same as above. This info is being extracted in the WorkspaceEventHandler with the assumption that the event originated from the Workspace.

Reply:
OK, so it'll always have an accgrp id in the case of the workspace, but the other two may not be present.